include/zephyr/bluetooth/mesh/access.h

```diff
@@ -521,11 +521,22 @@ struct bt_mesh_model_op {
 	BT_MESH_MODEL_UUIDS_UNASSIGNED()                                     \
 	.op = _op,                                                           \
 	.cb = _cb,                                                           \
-	.metadata = _metadata,                                               \
+	.metadata = __VA_ARGS__,                                             \
 }
 #else
-#define BT_MESH_MODEL_METADATA_CB(_id, _op, _pub, _user_data, _cb, _metadata)  \
```

Comment:
I guess this implementation will work too. You do not need to change it.

Reply:
I didn't get your point. Isn't it better to have clean self explanatory #define ?

Reply:
`BT_MESH_MODEL_METADATA_CB` is already a clean and self-explanatory definition by itself.
It is always worth to reuse existing code rather than to create new one. It reduces maintenance and minimizes the chance to create a new issue.

Reply:
I think suggestion in original discussion was to create single macros that will work both ways (with metadata or with NULL pointer). But now agreement is to have two macros?